Fix: Timezone dropdown label does not account for DST

niranjan-uma-shankar opened the pull request:
Fixes https://github.com/TryGhost/Ghost/issues/24559. The Admin facing changes are implemented in https://github.com/TryGhost/Ghost/pull/24557, and needs to be deployed along with this PR.

Changes:

* Remove the hardcoded GMT offsets in the label. 
* The `timezoneDataWithGMTOffset()` function dynamically prepends the GMT offset to the label, thereby ensuring that it respects DST. 
* Timezone data, after factoring in DST, is sorted by GMT offset to ensure that the dropdown values are intuitive.
* TS config updated to use es2020 which is supported by modern browsers and allows us to use `formatter.formatToParts()`.

troyciesco reviewed (dismissed):
@niranjan-uma-shankar before I dig too deep into this - could you add some tests for the new functions? Thanks!

niranjan-uma-shankar commented on packages/timezone-data/src/index.ts:
```diff
@@ -1,272 +1,330 @@
+};
+
+const labelWithGMTOffset = (label: string, offsetString: string): string => {
+    return '(' + offsetString + ') ' + label;
```
Makes sense, I've updated with this change.

niranjan-uma-shankar committed:
Added tests for timezone-data GMT offset helpers

Covers getGMTOffset() and timezoneDataWithGMTOffset() including DST
transitions, half-hour offsets and graceful failure, and updates the
existing static-data assertions now that labels no longer carry a
hardcoded GMT prefix.

niranjan-uma-shankar committed:
Made timezone-data GMT offsets DST-aware

- Removed the hardcoded (GMT ...) prefixes from the static labels, which
  were incorrect for half the year in any DST-observing zone
- Added getGMTOffset() and timezoneDataWithGMTOffset(), which derive the
  current offset at runtime via Intl and sort the list by offset
- Bumped the TS target to es2020 for formatToParts/padStart support

niranjan-uma-shankar committed:
Required Node 18+ for timezone-data

timezoneDataWithGMTOffset() relies on Intl's 'longOffset' timeZoneName,
which needs Node 18+ / modern ICU. On older runtimes the offset can't be
resolved and every label silently loses its GMT prefix, so make the floor
explicit.

9larsons reviewed (approved):
I fixed this up to work with the other PR(s) and decided we don't want to export the result as that fixes it in time.

9larsons committed:
Make GMT offset deterministic for UTC across ICU versions

The getGMTOffset() tests assumed Etc/UTC formats as a bare "GMT", but
newer Node/ICU builds emit "GMT+00:00", so on CI the offset string came
back as "GMT +0:00" and two assertions failed. It also left the !match
fallback branch uncovered (UTC no longer hit it), dropping line coverage
below the 100% threshold.

- Normalise any zero offset to a canonical "GMT" in getGMTOffset(), so
  UTC renders consistently as "(GMT) UTC" regardless of the runtime's
  Intl/ICU version (also fixes the latent UI inconsistency).
- Add a deterministic test that stubs formatToParts to exercise the
  raw-timeZoneName fallback branch, restoring 100% line coverage.
